crates/type-system/src/ontology/shared/all_of/repr.rs

```diff
@@ -0,0 +1,94 @@
+use serde::{Deserialize, Serialize};
+#[cfg(target_arch = "wasm32")]
+use tsify::Tsify;
+
+use crate::{repr, EntityTypeReference, ParseAllOfError};
+
+#[cfg_attr(target_arch = "wasm32", derive(Tsify))]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct AllOf<T> {
+    #[serde(
+        rename = "allOf",
+        default = "Vec::new",
```

Comment:
```suggestion
        default,
```

Reply:
This is needed because otherwise it expects `T` to implement default... I have no idea why

Reply:
Annoying, didn't expected that. 😞

Reply:
Me neither, I lose about 10 minutes trying to figure out why it wanted `Default` when I just want an empty vec 🙃 

It's not like we're even saying `default = "Vec<T>::new` so this is just arcane macro blackbox funtimes

Reply:
I thought this was because of https://github.com/rust-lang/rust/issues/26925 but the implementation is
```rust
impl<T> const Default for Vec<T> {
    /// Creates an empty `Vec<T>`.
    fn default() -> Vec<T> {
        Vec::new()
    }
}
```